src/iterator.rs

```diff
@@ -10,6 +10,17 @@ use super::*;
 
 use std::iter;
 
+/// Creates flat_tree Iterator
+///
+/// ## Examples
+/// ```rust
+/// use flat_tree::iterator;
+/// assert_eq!(iterator(0).take(3).collect::<Vec<usize>>(), [2, 4, 6]);
+/// ```
+pub fn iterator(index: usize) -> Iterator {
+  Iterator::new(index)
+}
+
```

Comment:
How is this different from `Iterator::new()` ? I'm thinking it might perhaps make more sense to add the doctests to `Iterator::new()` directly so we don't need to add more API surface :sparkles:

Reply:
Sure, I'll do that